sql.go

```diff
@@ -0,0 +1,56 @@
+package ksuid
+
+import (
+	"database/sql/driver"
+	"fmt"
+)
+
+// Scan implements the sql.Scanner interface. It supports converting from
+// string, []byte, or nil into a KSUID value. Attempting to convert from
+// another type will return an error.
+func (i *KSUID) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case nil:
+		return nil
+	case string:
+		// if an empty KSUID comes from a table, we return a null KSUID
+		if src == "" {
+			return nil
+		}
+		k, err := Parse(src)
+		if err != nil {
+			return fmt.Errorf("Scan: %v", err)
+		}
+		*i = k
+	case []byte:
+		// if an empty KSUID comes from a table, we return a null KSUID
+		if len(src) == 0 {
+			return nil
+		}
+		// assumes a simple slice of bytes if [byteLength] bytes
+		if len(src) == byteLength {
+			copy((*i)[:], src)
+			return nil
+		}
+
+		if len(src) == stringEncodedLength {
+			return i.Scan(string(src))
+		}
+
+		return i.Scan(string(src))
```

Comment:
These recursive calls to `Scan` cause heap allocations when packing the `string` value into an `interface{}`, this is why I have the unexported `scan` methods to use internally.

Reply:
@achille-roussel thanks for your time to review my request!

I see your Point. 

```suggestion
		return i.UnmarshalText(src)
```

Using the UnmarshalText is better. Both functions (scan old, scan new) are now allocating the same. Beside that the new Scan is consuming 2,79% less memory + is slightly faster.